exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -70,6 +70,7 @@ def initialize(endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPOR
                      else
                        raise ArgumentError, ERROR_MESSAGE_INVALID_HEADERS
                      end
+          @headers['User-Agent'] = "OTel OTLP Exporter Ruby/#{OpenTelemetry::Exporter::OTLP::VERSION}"
```

Comment:
This is the first time we're mutating the `headers` Hash that's passed in. In general, we assume the caller maintains ownership and doesn't want us stomping on their hash. Arguably, we should have `dup`ed the hash in the `when Hash` branch above, in case the caller subsequently mutated it. I think we should do that anyway, then we own the hash and this line is 👍 .

Reply:
```suggestion
          @headers['User-Agent'] = "OTel-OTLP-Exporter-Ruby/#{OpenTelemetry::Exporter::OTLP::VERSION}"
```

Product identifiers in a User-Agent come in [the format of a token](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6) and therefore should not include spaces, despite [the current spaced-string example provided in the OTel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/ff19b9e8212fae4296f63d307591d781d4613304/specification/protocol/exporter.md#user-agent). Spaces in this value are going to mess up a looooot of parsers.